### DISCUSSION TITLE
Receive images from Biim portable ultrasound probe

### DISCUSSION THREAD

**Discussion by lassoan** [PlusToolkit/PlusLib#792] [Q&A] [answered]
From https://github.com/DevanshShukla41:

17 hours ago
Hi Everyone!
We are working on a project in which we are using "Biim Portable Ultrasound probe," which uses a wifi connection (similar to Clarius Ultrasound Machine) that connects the probe to their mobile application to get the Ultrasound images.

We are going to interface it with the PLUS toolkit for data collection.

So for that, I wanted to ask you guys whether it will be possible to interface the PLUS toolkit for data collection with the "Biim Ultrasound Probe device," using the example which your lab has published on your website for Clarius Ultrasound Machine?

adamrankin
17 hours ago
Maintainer
Does Biim provide an API for receiving images over the network that isn't their app?

@DevanshShukla41
DevanshShukla41
16 hours ago
No Biim does not have an API for receiving the images over the network other than their mobile application, which is the problem.
But it does have an option in the application wherein we can basically input the server storage settings (Local AET Tittle; Remote AET Title; Server address; Port etc.) and send the dicom files on the server, and from there, I guess we could download them onto our PC'S.
I have attached the screenshots from the application for it below.

![image](https://user-images.githubusercontent.com/307929/109165529-a89b4c80-7749-11eb-994c-17278bd04235.png)

I am currently working on setting up an open-source server (EasyPacs) that helps me host the dicom images.

Do you think that could work?

Thank You so much, and I appreciate all your help!

**Comment by lassoan** [comment]
DICOM storage happens after the acquisition is completed. Even if automatic push is set up, you can expect variable time delay, in the magnitude of minutes. So, you will not be able to use DICOM transfer for volumetric reconstructions.

If you just want to see the images show up in 3D Slicer then you can enable the listening server in 3D Slicer (C-STORE SCP - see details [here](https://slicer.readthedocs.io/en/latest/user_guide/modules/dicom.html#dicom-networking)).

**Comment by adamrankin** [comment]
While Plus doesn't have a DICOM tie-in, 3D Slicer can connect to a PACS to read images. However, it will not be real-time and Plus won't be able to synchronize tracking and video timestamps.

**Comment by DevanshShukla41** [comment, marked as answer]
Thank You for the response, @lassoan, and @adamrankin.
I understand what you have mentioned above. Basically, from my understanding, we need a real-time video feed from the ultrasound device on our computer to synchronize PLUS to it. Biim Ultrasound device doesn't have an API that allows real-time video feed on the computer, so that would be the problem.
So if there is an API that allows getting the video feed on the computer, then that should work, right?
And is there something else that you guys think if I can figure out, then that could work for the Biim Ultrasound device?

Also, could you guys explain to me a bit how and why the Clarius Ultrasound Device is more helpful than the Biim Ultrasound Device? And are you able to synchronize PLUS with the Clarius device on the computer, or are you guys using the video feed from their mobile application via the Listener API that basically allows you to screen-cast the video feed from the mobile application to any other device?

Thank You so much! I really appreciate you guys helping me out!

**Reply by adamrankin** [reply]
> Also, could you guys explain to me a bit how and why the Clarius Ultrasound Device is more helpful than the Biim Ultrasound > Device? And are you able to synchronize PLUS with the Clarius device on the computer, or are you guys using the video feed > from their mobile application via the Listener API that basically allows you to screen-cast the video feed from the mobile > application to any other device?

Yes, that's exactly correct. Clarius has the Listener API that we can use to receive the live feed of video images.

**Reply by lassoan** [reply]
You can contact Biim device manufacturer to give you description of the communication protocol between the transducer and the app, but if you buy only a single device from them then probably it will not worth for them to deal with this.

If you want to get access to the real-time image stream then make sure you buy a device that actually offers this feature. For example, you can get a complete Telemed micrUS system for a few thousand $. If you need wireless then the only one I know that has real-time streaming interface is Clarius (but it is a pricier device and streaming access costs extra).
